### PR TITLE
Prepare 0.18.3 release

### PR DESCRIPTION
**Pull request by jakelishman**
### Summary

Bump version numbers for 0.18.3 release.  No fixes to current release notes are necessary this time (there's only one note, anyway).

### Details and comments

This version release is currently blocked on the following PRs being merged to `main`, and subsequently being backported to `stable/0.18`.

- [x] #7071 (backport PR #7074)
- [x] #7037 (backport PR #7077)
- [x] #7019 (backport PR #7076)
- [x] #7079

### CODE REVIEW

File: releasenotes/notes/prepare-0.18.3-321aa7dc38f23e29.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+prelude: >
+  This bugfix release fixes a few minor issues in the 0.18 series, including a
+  performance regression in :obj:`~qiskit.compiler.assemble` when dealing with
+  non-pulse backends.
```
Comment (mtreinish) [inline releasenotes/notes/prepare-0.18.3-321aa7dc38f23e29.yaml:5]: Just a small clarification it's a regression when we run a circuit on a pulse enabled ibm backend but aren't leveraging any of the pulse features:

```suggestion
  performance regression in :obj:`~qiskit.compiler.assemble` when dealing with
  executing :class:`~qiskit.circuit.QuantumCircuit` objects on pulse enabled
  backends.
```

**Comment by jakelishman** [comment]
All blockers are now merged, and we're ready for the release.

File: releasenotes/notes/prepare-0.18.3-321aa7dc38f23e29.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+prelude: >
+  This bugfix release fixes a few minor issues in the 0.18 series, including a
```
Comment (1ucian0) [inline releasenotes/notes/prepare-0.18.3-321aa7dc38f23e29.yaml:3]:
```suggestion
  This bugfix release fixes a few minor issues in 0.18, including a
```

**Review by 1ucian0 — DISMISSED** [review]

**Comment by mtreinish** [comment]
Before we merge this we need to check the release notes. For example there is a typo here: https://github.com/Qiskit/qiskit-terra/blob/stable/0.18/releasenotes/notes/sx-error-map-408f28a6a0c05fd2.yaml#L6 Let's check the output in the docs ci job and make sure it all looks good before we merge this and tag the release

**Comment by jakelishman** [comment]
Ah yeah, of course the new notes that I blocked this release weren't in my branch when I did the build test earlier.  I'd checked all the incoming ones in text form when I wrote them in their respective PRs, but must have missed the `` ` ' `` transposition.

**Review by mtreinish — APPROVED** [review]

**Pull request merged**